src/GitVersion.Core.Tests/IntegrationTests/CreatingAFeatureBranchFromAReleaseBranchScenario.cs

```diff
@@ -0,0 +1,701 @@
+        fixture.AssertFullSemver("1.0.0+0", configuration);
+
+        fixture.ApplyTag("1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0", configuration);
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configuration);
+
+        fixture.MergeNoFF("main");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.6", configuration);
+
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ❌ expected: "1.1.0-alpha.2" because only one commit and one merge has been pushed
+        fixture.AssertFullSemver("1.1.0-alpha.6", configuration);
+
+        fixture.Repository.DumpGraph();
+    }
+
+    [Test]
+    public void ShouldNotConsiderTheMergeCommitFromReleaseToMainWhenCommitHasNotBeenTagged()
+    {
+        // *457e0cd in the future(HEAD -> develop)
+        // |\  
+        // | *d9da657 in the future(tag: 1.0.0, main)
+        // | |\  
+        // | | *026a6cd in the future
+        // | | *7f5de6e in the future
+        // * | | 3db6e6f in the future
+        // | |/
+        // |/|
+        // * | 845926e in the future
+        // |/
+        // *42db9ba in the future
+
+        var configuration = TestConfigurationBuilder.New.Build();
+
+        using var fixture = new EmptyRepositoryFixture();
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.0.1+1", configuration);
+
+        fixture.BranchTo("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("0.1.0-alpha.2", configuration);
+
+        fixture.BranchTo("release/1.0.0");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+1", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.0.0-beta.1+2", configuration);
+
+        fixture.Checkout("develop");
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.0", configuration);
+
+        fixture.MakeACommit();
+
+        // ✅ succeeds as expected
+        fixture.AssertFullSemver("1.1.0-alpha.1", configuration);
+
+        fixture.Checkout("main");
+        fixture.MergeNoFF("release/1.0.0");
+
+        // ❌ expected: "0.0.1+4" because until the commit is not tagged it's a hotfix
+        fixture.AssertFullSemver("1.0.0+0", configuration);
+
+        fixture.Repository.Branches.Remove("release/1.0.0");
+
+        // ❌ expected: "0.0.1+4" because until the commit is not tagged it's a hotfix
+        fixture.AssertFullSemver("1.0.0+4", configuration);
```

Comment:
Yes that goes in the same direction. In my opinion it depends on the VersionMode or other configurable properties like track-merge-message. You can think of that some users are expecting this as hotfix because the release branch has been deleted and this might be the same like this has never been existing.